PopPUNK/__main__.py

```diff
@@ -142,12 +143,26 @@ def get_options():
 
     # qc options
     qcGroup = parser.add_argument_group('Quality control options')
-    qcGroup.add_argument('--max-a-dist', default = 0.5, type=float, help='Maximum accessory distance to permit '
-                                                                         '[default = 0.5]')
-    qcGroup.add_argument('--ignore-length', help='Ignore outliers in terms of assembly length '
-                                                 '[default = False]', default=False, action='store_true')
-    qcGroup.add_argument('--estimated-length', default=2000000, type = int, help='Provide an integer estimated genome length when using "--ignore-length" [default = 2000000]')
-
+    qcGroup.add_argument('--qc-filter', help='Behaviour following sequence QC step: "stop" [default], "prune"'
+                                                '(analyse data passing QC), or "continue" (analyse all data)',
+                                                default='stop', type = str)
+    qcGroup.add_argument('--retain-failures', help='Retain sketches of genomes that do not pass QC filters'
+                                                '[default = False]', default=False, action='store_true')
+    qcGroup.add_argument('--max-a-dist', help='Maximum accessory distance to permit [default = 0.5]',
+                                                default = 0.5, type = float)
+    qcGroup.add_argument('--estimated-length', help='Provide an integer estimated genome length for QC',
+                                                default=None, type = int)
+    qcGroup.add_argument('--length-sigma', help='Number of standard deviations of length distribution beyond '
+                                                'which sequences will be excluded [default = 5]', default = 5, type = int)
+    qcGroup.add_argument('--lower-length', help='Lower length below which sequences will be excluded',
+                                                default=None, type = int)
+    qcGroup.add_argument('--upper-length', help='Upper length above which sequences will be excluded',
+                                                default=None, type = int)
```

Comment:
I would suggest replacing with `--allowed-length` with `nargs=2` to give the range, also avoids the check below

Reply:
Good idea, done in 5fd8f0e, also included in new test added in b039242.